server/service.go

```diff
@@ -646,6 +651,143 @@ func (m *BoostService) processCapellaPayload(w http.ResponseWriter, req *http.Re
 	m.respondOK(w, result)
 }
 
+func (m *BoostService) processDenebPayload(w http.ResponseWriter, req *http.Request, log *logrus.Entry, blindedBlock *eth2ApiV1Deneb.SignedBlindedBeaconBlock) {
```

Comment:
would it be worth making a generic processPayload function that works with both deneb and capella blocks? just in terms of reducing code copying for ease of review? maybe not if we plan on stripping out the capella payload processing function immediately after the fork. just a thought as its hard to tell what logic is new with deneb and which is just copied from capella.

Reply:
I think we're going to strip Capella shortly afterwards.

Reply:
I've thought about this approach but there are issues as the process payload function accepts different types. We would need to have a custom interface to extract the common fields and process them which leads to more potential bugs.w/ versioning. There is code duplication but less prone to issues with version handling.

Reply:
good point @avalonche, agree. this is a space with high potential for critical bugs, better be safe with some duplication